src/Runner.Worker/Handlers/CompositeActionOutputHandler.cs

```diff
@@ -0,0 +1,77 @@
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using GitHub.DistributedTask.ObjectTemplating.Schema;
+using GitHub.DistributedTask.ObjectTemplating.Tokens;
+using GitHub.DistributedTask.Pipelines.ContextData;
+using GitHub.DistributedTask.WebApi;
+using GitHub.Runner.Common;
+using GitHub.Runner.Sdk;
+using Pipelines = GitHub.DistributedTask.Pipelines;
+
+namespace GitHub.Runner.Worker.Handlers
+{
+    [ServiceLocator(Default = typeof(CompositeActionOutputHandler))]
+    public interface ICompositeActionOutputHandler : IHandler
+    {
+        CompositeActionExecutionData Data { get; set; }
+    }
+
+    public sealed class CompositeActionOutputHandler : Handler, ICompositeActionOutputHandler
+    {
+        public CompositeActionExecutionData Data { get; set; }
+
+
+        public Task RunAsync(ActionRunStage stage)
+        {
+            // Evaluate the mapped outputs value
+            if (Data.Outputs != null)
+            {
+                // We have to have this function because ExecutionContext.ToPipelineTemplateEvaluator calls
+                // PipelineTemplateSchemaFactory.GetSchema() which always uses workflow_v1.0.json 
+                // So, we have to manually create the evaluator
+                var evaluator = new Pipelines.ObjectTemplating.PipelineTemplateEvaluator(ExecutionContext.ToTemplateTraceWriter(), ExecutionContext.ActionManifestSchema, ExecutionContext.FinalizeContext.FileTable);
+
+                // Evaluate the outputs in the steps context to easily retrieve the values
+                DictionaryContextData actionOutputs = evaluator.EvaluateCompositeOutputs(Data.Outputs, ExecutionContext.ExpressionValues, ExecutionContext.ExpressionFunctions);
+
+                // Each pair is structured like this
```

Comment:
Since comment intended for conceptual understanding of the structure, and not how the object is serialized, consider plain json dictionaries to describe the structure instead

```
{
  "the-output-name": {
    "description": "",
    "value": "the value"
  },
  ...
}
```